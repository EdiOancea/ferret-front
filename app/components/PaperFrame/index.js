import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = theme => ({
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme
      .spacing.unit * 3}px`,
  },
  avatar: {
    margin: theme.spacing.unit,
    backgroundColor: theme.palette.secondary.main,
  },
});

const PaperFrame = props => {
  const { classes, text, icon, children } = props;

  return (
    <Paper className={classes.paper}>
      <Avatar className={classes.avatar}>{icon}</Avatar>
      <Typography component="h1" variant="h5">
        {text}
      </Typography>
      {children}
    </Paper>
  );
};

PaperFrame.propTypes = {
  classes: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  children: PropTypes.object,
};

export default withStyles(styles)(PaperFrame);
