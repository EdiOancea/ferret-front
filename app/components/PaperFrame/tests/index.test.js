import React from 'react';
import { render } from 'enzyme';
import PaperFrame from 'components/PaperFrame';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';

describe('<PaperFrame />', () => {
  it('should not regress', () => {
    const tree = render(
      <PaperFrame text="test text" icon={<LockOutlinedIcon />}>
        <div>test</div>
      </PaperFrame>,
    );
    expect(tree.children.length).toEqual(1);
    expect(tree).toMatchSnapshot();
  });
});
