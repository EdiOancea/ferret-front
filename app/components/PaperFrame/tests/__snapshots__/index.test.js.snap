// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaperFrame /> should not regress 1`] = `
<div
  class="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 PaperFrame-paper-1"
>
  <div
    class="MuiAvatar-root-30 MuiAvatar-colorDefault-31 PaperFrame-avatar-2"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-33"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <g
        fill="none"
      >
        <path
          d="M0 0h24v24H0V0z"
        />
        <path
          d="M0 0h24v24H0V0z"
          opacity=".87"
        />
      </g>
      <path
        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
      />
    </svg>
  </div>
  <h1
    class="MuiTypography-root-42 MuiTypography-h5-58"
  >
    test text
  </h1>
  <div>
    test
  </div>
</div>
`;
